From c2a5e05231ea0743ef89e666c37a0a77534b9573 Mon Sep 17 00:00:00 2001
From: Michael Catanzaro <mcatanzaro@gnome.org>
Date: Thu, 07 Nov 2013 03:53:45 +0000
Subject: NautilusFile: fix crash in modify_link_hash_table

We crash in g_hash_table_lookup_extended if the file's symlink_name is
set but the file is not a symlink, since the hash function (g_str_hash)
does not allow NULL values but target_uri is NULL.

https://bugzilla.gnome.org/show_bug.cgi?id=711583
---
Index: nemo-3.8.2/libnemo-private/nemo-file.c
===================================================================
--- nemo-3.8.2.orig/libnemo-private/nemo-file.c	2014-01-15 14:32:11.321604804 +1300
+++ nemo-3.8.2/libnemo-private/nemo-file.c	2014-01-15 14:32:11.317604804 +1300
@@ -556,7 +556,7 @@
 	GList **list_ptr;
 
 	/* Check if there is a symlink name. If none, we are OK. */
-	if (file->details->symlink_name == NULL) {
+	if (file->details->symlink_name == NULL || !nemo_file_is_symbolic_link (file)) {
 		return;
 	}
 
